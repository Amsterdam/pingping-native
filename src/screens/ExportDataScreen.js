import React from 'react';
import {StyleSheet, TextInput, View} from 'react-native';
import {Content, Container} from 'native-base';
import ContentLayout from '../components/layout/ContentLayout';
import LabeledHeader from '../components/header/LabeledHeader';
import Title from '../components/typography/Title';
import Body from '../components/typography/Body';
import Button from '../components/OnboardingButton';

const styles = StyleSheet.create({
  input: {
    height: 40,
    alignSelf: 'stretch',
    borderColor: '#000',
    borderWidth: 1,
    marginBottom: 10,
    paddingLeft: 10,
    paddingRight: 10,
  },
  margin: {
    marginBottom: 30,
  },
  inputContainer: {alignItems: 'center', justifyContent: 'center'},
});

const ImportDataScreen = ({navigation}) => {
  const [value, onChangeText] = React.useState('');
  return (
    <Container>
      <LabeledHeader filledHeader navigation={navigation} title="Profiel" />
      <Content>
        <ContentLayout>
          <Title style={styles.margin}>Gegevens Exporteren</Title>
          <Body style={styles.margin}>
            Als je van device switcht wil je natuurlijk niet dat al jouw
            gegevens en prestaties op Ping Ping verloren gaan!
          </Body>
          <Body style={styles.margin}>
            Het is heel simpel om jouw gegevens te exporteren naar een nieuw
            device. Kopieer de UUID (Universally Unique Identifier) hieronder en
            plak hem in de 'importeer gegevens' pagina op jouw nieuwe device!
          </Body>
          <View style={styles.inputContainer}>
            <TextInput
              style={styles.input}
              onChangeText={(text) => onChangeText(text)}
              value={value}
            />
            <Button
              rounded
              disabled={!value}
              label="Token Kopieren"
              iconType="AntDesign"
              iconName="copy1"
              onPress={() => console.log('copy this stuff')}
            />
          </View>
        </ContentLayout>
      </Content>
    </Container>
  );
};

export default ImportDataScreen;
