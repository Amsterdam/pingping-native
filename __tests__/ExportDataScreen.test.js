import React from 'react';

import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';

import ExportDataScreen from '../src/screens/ExportDataScreen';

jest.useFakeTimers();
const navigation = { navigate: jest.fn() };

test('renders correctly', () => {
	const tree = renderer
		.create(
			<MockedProvider mocks={[]}>
				<ExportDataScreen navigation={navigation} />
			</MockedProvider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
