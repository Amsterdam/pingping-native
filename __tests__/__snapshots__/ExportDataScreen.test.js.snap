// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#0d2036",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0d2036",
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 56,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "left": 5,
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="HEADER_BACK_BUTTON"
      >
        <RNSVGSvgView
          bbHeight={24}
          bbWidth={24}
          fill="none"
          focusable={false}
          height={24}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              clipRule={0}
              d="M19.7 11H7.5l5.6-5.6L11.7 4l-8 8 8 8 1.4-1.4L7.5 13h12.2v-2Z"
              fill={4294967295}
              fillRule={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={0}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Heavitas",
          },
          Object {
            "fontSize": 26,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#fff",
            "marginTop": 5,
          },
        ]
      }
    >
      Profiel
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 25,
            "position": "relative",
            "width": 750,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Heavitas",
              },
              Object {
                "fontSize": 58,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Gegevens Exporteren
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Raleway-Regular",
                "lineHeight": 46,
              },
              Object {
                "fontSize": 29,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Als je van device switcht wil je natuurlijk niet dat al jouw gegevens en prestaties op Ping Ping verloren gaan!
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Raleway-Regular",
                "lineHeight": 46,
              },
              Object {
                "fontSize": 29,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Het is heel simpel om jouw gegevens te exporteren naar een nieuw device. Open op je nieuwe device de app en klik rechtsboven op inloggen. Vervolgens scan je de onderstaande QRCode en zo simpel is het!
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#fb9f00",
                "borderRadius": 5,
                "borderWidth": 5,
                "padding": 10,
              }
            }
          >
            <ActivityIndicator
              color="#fb9f00"
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
